src/CoreWCF.Primitives/src/CoreWCF/Pool.cs

```diff
@@ -3,7 +3,7 @@
 
 namespace CoreWCF
 {
-    internal class Pool<T> where T : class
+    public class Pool<T> where T : class
     {
```

Comment:
This class really needs to go away. Can you replace usage of this with `Microsoft.Extensions.ObjectPool.DefaultObjectPool<T>`. There's a usage of this in AsyncLock is you want to see how it's used.

Reply:
I'm having some trouble swapping this out. The buffer is null when the pool is initialized so it will stay that way. This will cause a null argument exception when an instance is retrieved from the pool. Ignoring the buffer and providing an empty array will work for the JSON side of things, but fails for the XML side of things (EOF failure). I have this code stashed, maybe we can link up about this at some point.

Reply:
I think I have this working, but it looks a little wonky for the XML paths. I initialize the readers with a tiny XML string which gets swapped out a bit later in a call. It seems to work, but is a bit hacky. Commit in question is here: 87030bffdd5728de3564032c8b9511baace01e1b.

Reply:
This is causing the tests to fail on Linux. Weirdly they pass on Windows.

Reply:
This is finally resolved.